packages/flutter_tools/lib/src/android/android_emulator.dart

```diff
@@ -0,0 +1,115 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+import '../android/android_sdk.dart';
+import '../android/android_workflow.dart';
+import '../base/file_system.dart';
+import '../base/process.dart';
+import '../emulator.dart';
+import '../globals.dart';
+import 'android_sdk.dart';
+
+class AndroidEmulators extends EmulatorDiscovery {
+  @override
+  bool get supportsPlatform => true;
+
+  @override
+  bool get canListAnything => androidWorkflow.canListEmulators;
```

Comment:
I think we can remove this method for the emulators - this is used for devices to know whether we should display an error in flutter doctor; we won't need that for emulators.

Reply:
Ah, I see you use it below.

Reply:
Yeah, if we can't find `emulator` or the AVD folder, we output a message.